src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1599,19 +1602,20 @@ private SolutionCompilationState ComputeFrozenSnapshot(CancellationToken cancell
                 if (oldTracker == newTracker)
                     continue;
 
-                Contract.ThrowIfFalse(newIdToProjectStateMapBuilder.ContainsKey(projectId));
-
-                var newProjectState = newTracker.ProjectState;
-
-                newIdToProjectStateMapBuilder[projectId] = newProjectState;
+                newProjectStatesBuilder[i] = newTracker.ProjectState;
                 newIdToTrackerMapBuilder[projectId] = newTracker;
+                projectStateChanged = true;
             }
```

Comment:
going to trust that sorting stayes the same.  i think so, but it might be worthwhile to doc.  since we're not changing IDs, the order would stay the same.